digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -79,7 +80,15 @@ class RoadAddressService(roadLinkService: RoadLinkService) {
     val viiteRoadLinks = roadLinks.flatMap { rl =>
       val ra = addresses.getOrElse(rl.linkId, Seq())
       buildRoadAddressLink(rl, ra)
+  }
+    val newAnomalousRL = viiteRoadLinks.filter(rl => (rl.administrativeClass == State || rl.roadNumber > 0) && rl.id == 0)
+
+    val missedRL  = withDynTransaction {
+    RoadAddressDAO.getMissingRoadAddresses()
```

Comment:
1. I think is better to get only the roadAddresses with link id inside the viiteRoadLinks. Instead of getting the all table.
2. I belive this is not releated with this task but the RoadAddressDAO shouldn't be a class so we can mock it on the unit tests?